packages/ai-jsx/src/core/conversation.tsx

```diff
@@ -215,7 +217,21 @@ export async function renderToConversation(
     ) : (
       conversation
     );
-  return toConversationMessages(await render(conversationToUse, { stop: isConversationalComponent }));
+  const messages = toConversationMessages(await render(conversationToUse, { stop: isConversationalComponent }));
+
+  if (logger && logType) {
+    const loggableMessages = await Promise.all(
+      messages.map(async (m) => ({
+        element: debug(m.element, true),
+        ...(cost && { cost: await cost(m, render) }),
+      }))
+    );
+
+    logger.setAttribute(`ai.jsx.${logType}`, JSON.stringify(loggableMessages));
+    logger.info({ [logType]: { messages: loggableMessages } }, `Got ${logType} conversation`);
```

Comment:
Are there links you can share of how this looks?

Reply:
ah nm I see you shared https://app.datadoghq.com/apm/trace/14113410336604669720?graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=15969530707541520513 in a separate comment